### PR TITLE
Add external resource provider to USB Host Pass through device

### PR DESCRIPTION
**Pull request by victortoso**
**What this PR does / why we need it**:

This is a follow up from https://github.com/kubevirt/kubevirt/pull/10015
It does minor fixes suggested by @xpivarc and adds support to `external resource provider`, similar to PCI and Mediated devices.

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by victortoso** [comment]
`pull-kubevirt-e2e-k8s-1.28-sig-storage`

```
[Serial][sig-compute][USB] host USB Passthrough with usb storage with emulated USB devices [It] Should successfully passthrough 1 emulated USB device [sig-compute, USB]
```

IIUC, the test should not run in the sig-storage, only in sig-compute. Perhaps some logic is missing wrt to `USB` decorator

File: pkg/virt-handler/device-manager/usb_device.go [hunk 1]
```diff
@@ -605,6 +605,11 @@ func discoverAllowedUSBDevices(usbs []v1.USBHostDevice) map[string][]*PluginDevi
 	localDevices := discoverLocalUSBDevicesFunc()
 	for _, usbConfig := range usbs {
 		resourceName := usbConfig.ResourceName
+		if usbConfig.ExternalResourceProvider {
+			log.Log.V(4).Infof("Skipping discovery of %s. To be handled by other device-plugin",
```
Comment (xpivarc) [inline pkg/virt-handler/device-manager/usb_device.go:609]: nit: "by other" -> "by external".
I would not be opposed to increasing the verbosity even more.

**Review by xpivarc — COMMENTED** [review]
/approve

File: pkg/virt-handler/device-manager/usb_device.go (hunk 1 above)
Comment (victortoso) [reply pkg/virt-handler/device-manager/usb_device.go:609]: I'll move from 4 to 6, but we do need to define some [convention](https://github.com/kubevirt/kubevirt/issues/8164) eventually.

Thanks for the quick review

Comment (xpivarc) [reply pkg/virt-handler/device-manager/usb_device.go]: I think such guidance would be nice.

**Comment by victortoso** [comment]
/retest

```
Error: netavark: code: 3, msg: modprobe: FATAL: Module ip_tables not found in directory /lib/modules/5.14.0-200.el9.aarch64
```

**Comment by victortoso** [comment]
/test pull-kubevirt-unit-test-arm64

```
 Summarizing 1 Failure:
  [FAIL] virt-handler retry manager [It] Should do exponential backoff on failure retries
  pkg/virt-handler/retry_manager_test.go:62 
```

**Pull request merged**